src/osd/osd_types.cc

```diff
@@ -4668,21 +4669,58 @@ void pg_log_t::copy_after(const pg_log_t &other, eversion_t v)
     }
     log.push_front(*i);
   }
+  for (auto i = other.log.cbegin(); i != other.log.cend(); ++i) {
```

Comment:
it looks like the trim portion for both of these is the same, and could extracted into a common helper